test/opengl/save_state.cpp

```diff
@@ -7,47 +7,21 @@ int main() {
 }
 #else
 
-#include <cstring>
-#include <stddef.h>
-#include <stdio.h>
-#include <stdlib.h>
-
 #include "Halide.h"
 
-#include "runtime/mini_opengl.h"
```

Comment:
This change seems orthogonal to the rest of the PR? The decision to not rely on the existence of functioning GPU API headers is intentional.

Reply:
I understand that for the runtime, but why should the tests have to access to the internal runtime headers? It intersects with this PR since the CMake build detects the appropriate paths for functioning GPU API headers anyway, and this test wouldn't build without exposing the runtime headers to the tests.

Furthermore, since the change works with the Makefile build, where tested (only Linux), I'm not sure why we _shouldn't_ use a system header for a test.

All this is moot if we end up dropping OpenGL support, though. It isn't tested on Windows (where it doesn't even work) or macOS (where it also doesn't even work #4885 )

Reply:
Reasonable points (and it probably *is* moot), but I didn't see how it was related to this PR, which is why I commented. 

Is the reason it's in this PR because the tests no longer have access to internal runtime headers? That would make sense.

Reply:
I believe it worked at one point in time on all platforms. It wasn't added to the buildbots, so it rotted.

Reply:
> Is the reason it's in this PR because the tests no longer have access to internal runtime headers? That would make sense.

Yes. `mini_opengl.h` is in `src/runtime`, so it's not visible to the tests anymore.